### PR TITLE
Add institutional linking module

### PR DESCRIPTION
**Pull request by DanielPalafox**
https://trello.com/c/LfCTI7Wu/6682-social-institutional-institutional-link-account-8

### CODE REVIEW

**Comment by DanielPalafox** [comment]
[Social/institutional - institutional link account (8)](https://trello.com/c/LfCTI7Wu/6682-social-institutional-institutional-link-account-8)

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { Institutional } from '../../../types/institutional.endpoint'
+
+@Component({
+  selector: 'app-institutional-linking',
+  templateUrl: './institutional-linking.component.html',
+  styleUrls: [
+    './institutional-linking.component.scss',
+    './institutional-linking.component.scss-theme.scss',
+  ],
+  host: { class: 'container' },
```
Comment (leomendoza123) [inline src/app/link-account/pages/institutional-linking/institutional-linking.component.ts:16]: The use of this host is against the angular style guide https://angular.io/guide/styleguide#style-06-03)

This was supposed to be caught by the linter but it seems like at some update the migration failed to update some rules.

you can go and change these 3 lines 

https://github.com/ORCID/orcid-angular/blob/46abde39c7a1905bb7ee11c32070025486099368/tslint.json#L88-L90

with this 

```
    "no-inputs-metadata-property": true,
    "no-outputs-metadata-property": true,
    "no-host-metadata-property": true,
```

so the linter catch this thing again

Comment (leomendoza123) [reply src/app/link-account/pages/institutional-linking/institutional-linking.component.ts:16]: More info about the change on this rules names https://github.com/mgechev/codelyzer/issues/791#issuecomment-474031004

File: src/app/sign-in/sign-in.module.ts [hunk 2]
```diff
@@ -39,5 +41,11 @@ import { FormDirectivesModule } from '../cdk/form-directives/form-directives.mod
     FormDirectivesModule,
   ],
   entryComponents: [],
+  exports: [
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/sign-in/sign-in.module.ts:44]: Great catch!. Yes, that's right I will remove them

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    this.loadShibbolethSignInData()
+  }
+
+  ngOnInit(): void {
+    this.authorizationForm = new FormGroup({})
+  }
+
+  loadShibbolethSignInData() {
+    this._oauthService.loadShibbolethSignInData()
+      .pipe(take(1))
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/link-account/pages/institutional-linking/institutional-linking.component.ts:40]: ok

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.html [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          account to your ORCID record
+        </ng-container>
+      </mat-card-title
+      >
+    </mat-card-header>
+    <mat-progress-bar *ngIf="loading" mode="indeterminate"></mat-progress-bar>
+    <mat-card-content>
+      <div
+        class="form-container"
+        *ngIf="shibbolethSignInData && !shibbolethSignInData.unsupportedInstitution && !shibbolethSignInData.unsupportedInstitution && loadedFeed"
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/link-account/pages/institutional-linking/institutional-linking.component.html:19]: Great Catch! I will remove it

File: src/app/sign-in/components/form-sign-in/form-sign-in.component.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+import { Router } from '@angular/router'
+
+@Component({
+  selector: 'app-form-sign-in',
+  templateUrl: './form-sign-in.component.html',
+  styleUrls: ['./form-sign-in.component.scss'],
+  providers: [TwoFactorComponent],
+})
+export class FormSignInComponent implements OnInit {
+  @Input() signInType: string
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/sign-in/components/form-sign-in/form-sign-in.component.ts:17]: Ok

File: src/app/core/sign-in/sign-in.service.ts [hunk 6]
```diff
@@ -12,16 +12,25 @@ import { getOrcidNumber } from '../../constants'
...
-  signIn(data) {
-    const headers = new HttpHeaders().set(
+  ) {
+    this.headers = new HttpHeaders().set(
       'Content-Type',
       'application/x-www-form-urlencoded;charset=utf-8'
     )
+  }
+
+  signIn(data, type) {
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/core/sign-in/sign-in.service.ts:27]: Done

File: src/app/core/oauth/oauth.service.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        'Access-Control-Allow-Origin': '*',
+        'Content-Type': 'application/json'
+      }
+    );
+  }
+
+  loadShibbolethSignInData( ): Observable<ShibbolethSignInData> {
+    return this._http.get<ShibbolethSignInData>(
+      environment.BASE_URL + 'shibboleth/signinData.json',
+      { headers: this.headers }
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/core/oauth/oauth.service.ts:29]: Ok

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.html [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            <ng-container i18n="@@linking.button.1">
+              Sign in and link your
+            </ng-container>
+            {{institutionName}}
+            <ng-container i18n="@@linking.button.2">
+              account
+            </ng-container>
+          </button>
+          <button
+            mat-raised-button
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/link-account/pages/institutional-linking/institutional-linking.component.html:75]: Ok

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.ts (hunk 1 above)
Comment (DanielPalafox) [reply src/app/link-account/pages/institutional-linking/institutional-linking.component.ts]: Ok

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.scss [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+mat-card-title {
+  max-width: 448px;
+}
+
+button {
+  width: 100%;
+  margin-top: 12px;
+  border: solid 1px rgba(0, 0, 0, 0.12);
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/link-account/pages/institutional-linking/institutional-linking.component.scss:12]: Ok, I will obtain the color from the palette

File: src/app/core/sign-in/sign-in.service.ts [hunk 10]
```diff
@@ -12,29 +12,38 @@ import { getOrcidNumber } from '../../constants'
...
-  signIn(data) {
-    const headers = new HttpHeaders().set(
+  ) {
+    this.headers = new HttpHeaders().set(
       'Content-Type',
       'application/x-www-form-urlencoded;charset=utf-8'
     )
+  }
+
+  signIn(SingInLocal) {
```
Comment (leomendoza123) [inline src/app/core/sign-in/sign-in.service.ts:27]: Add the new type here so this function gets the definition

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.html [hunk 11]
```diff
@@ -1,86 +1,105 @@
...
-          <button
-            mat-raised-button
-            routerLink="/signin"
-            class="row mat-elevation-z0 display-1"
-            i18n="@@linking.cancel"
-          >
-            Cancel and go back
-          </button>
-        </app-form-sign-in>
+<main>
```
Comment (leomendoza123) [inline src/app/link-account/pages/institutional-linking/institutional-linking.component.html:1]: We need to add an `id="main"` on this main tag to allow the i18y url to work. 

This is the card about a QA error related to this https://trello.com/c/Oj2smo2Q/6777-qa-skip-to-main-content-doesnt-work-in-angular-pages

This is a fix PR to fix this in all other features https://github.com/ORCID/orcid-angular/pull/228

File: src/app/core/sign-in/sign-in.service.ts (hunk 10 above)
Comment (leomendoza123) [inline src/app/core/sign-in/sign-in.service.ts:27]: One final change here... let's add the type you have defined for this one

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.html [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
```
Comment (leomendoza123) [inline src/app/link-account/pages/institutional-linking/institutional-linking.component.html:1]: We need to add 'id=main' here 

We have a related QA error https://trello.com/c/Oj2smo2Q/6777-qa-skip-to-main-content-doesnt-work-in-angular-pages
And a fix to add this in all other features https://github.com/ORCID/orcid-angular/pull/228

File: src/app/core/sign-in/sign-in.service.ts (hunk 10 above)
Comment (DanielPalafox) [reply src/app/core/sign-in/sign-in.service.ts:27]: Done

Comment (DanielPalafox) [reply src/app/core/sign-in/sign-in.service.ts:27]: Done

File: src/app/link-account/pages/institutional-linking/institutional-linking.component.html (hunk 11 above)
Comment (DanielPalafox) [reply src/app/link-account/pages/institutional-linking/institutional-linking.component.html:1]: Ok, I will add it

**Comment by DanielPalafox** [comment]
https://github.com/ORCID/registry_vagrant/pull/567
https://github.com/ORCID/ORCID-Source/pull/5883

**Pull request merged**